### ISSUE TITLE
Problems building node addon when electron-prebuilt is pinned to a specific release

### ISSUE THREAD

**Issue by tysenmoore-xse** [szwacz/electron-boilerplate#192]
I am using the boilerplate and I've added node addons (ref, ref-array, etc).  However, I find that when I tried to pin electron-prebuilt to a specific release (i.e. "1.2.5" instead of "^1.2.5") I get the errors below.  

```
> ref@1.3.2 install /home/swdev/projects/app/node_modules/ref
> node-gyp rebuild

gyp info it worked if it ends with ok
gyp info using node-gyp@3.3.1
gyp info using node@6.3.0 | linux | x64
gyp ERR! configure error 
gyp ERR! stack Error: Invalid version number: .2.5
gyp ERR! stack     at getNodeDir (/home/swdev/projects/node_modules/npm/node_modules/node-gyp/lib/configure.js:69:25)
gyp ERR! stack     at /home/swdev/projects/node_modules/npm/node_modules/node-gyp/lib/configure.js:41:7
gyp ERR! stack     at /home/swdev/projects/node_modules/npm/node_modules/node-gyp/lib/configure.js:393:9
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:193:7)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at maybeClose (internal/child_process.js:852:16)
gyp ERR! stack     at Socket.<anonymous> (internal/child_process.js:323:11)
gyp ERR! stack     at emitOne (events.js:96:13)
gyp ERR! stack     at Socket.emit (events.js:188:7)
gyp ERR! System Linux 3.13.0-92-generic
gyp ERR! command "/usr/local/bin/node" "/home/swdev/projects/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/swdev/projects/app/node_modules/ref
gyp ERR! node -v v6.3.0
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok 
npm info lifecycle ref@1.3.2~install: Failed to exec install script
```

It seems node-gyp is truncating the version from my package.json.  But what is not clear to me is where it gets that version to begin with or why it cares.  I'd like to report a bug somewhere but I am not sure if it is me, the boilerplate interactions, node-gyp, etc.  Has anyone else experience the same problem or know what might be the problem?  Thanks.

**Comment by black-snow** [comment]
Which version of the boilerplate are you using?

**Comment by tysenmoore-xse** [comment]
I started development pre-1.0.0 and have tried to keep as up to date with changes as possible.  I also have the latest versions of all the npm packages.

After your question I did go back and try to reproduce the problem with 3.0.0 and 1.0.0 and had no success.  It seems this is something pre-1.0.0 or something I've introduced.  I will have to go through the install task in greater detail to figure this one out.  If anyone has any ideas as to why this would be a problem I'd appreciate any input.  In the meantime I will close the issue, if I figure out the problem I will post the solution for others.
